src/Block.tsx

```diff
@@ -0,0 +1,44 @@
+import * as React from 'react';
+
+interface IBlockProps {
+  code: string;
+  good: boolean;
+  language: string;
+}
+
+enum Highlight {
+  None = 1,
+  Good,
+  Bad,
+}
+
+function highlightCssClass(h: Highlight) {
+  return Highlight[h].toLowerCase();
+}
+
+interface IBlockState {
+  hl: Highlight;
+}
+
+class Block extends React.Component<IBlockProps, IBlockState> {
+  constructor(props: IBlockProps) {
+    super(props);
+    this.state = { hl: Highlight.None };
+  }
+
+  public render() {
+    const langclass = "language-" + this.props.language;
+    const classes = langclass + " block " + highlightCssClass(this.state.hl);
+    return (<code className={classes} onClick={this.click}>
+      {this.props.code}
+    </code>);
+  }
+
+  private click = (e: any) => {
```

Comment:
Just curious: why is this in arrow function form rather than a named function?

Reply:
named function doesn't bind to this, look up "typescript bind this"